### ISSUE TITLE
1.0 Roadmap

### ISSUE THREAD

**Issue by riperiperi** [riperiperi/Simitone#8]
If we can get Simitone to a good stage, I'll feel comfortable making an "official" release and writing a ton about it on the blog. There's quite a way to go for the game to be this complete, but it's pretty close to completion and the goal is in sight.

This goes into a lot of detail. It's likely that multiple of these issues could be covered in one day of development, though obviously very far from every day will be spent on development. ;)

This issue is subject to change, it's pretty much a draft right now. There's a lot to think about, so I probably easily missed something.

# Must have for launch:

## UI
- [ ] Options Menu
- [ ] Credits (OG credits and FreeSO contributors)
- [ ] UI for Spellbook and Cookbook.
- [ ] UI for purchasing clothing and pets. #7
- [ ] Cross-System game file transfer (ASP.NET OWIN server app + downloader UI) for transferring installations to laptops without disk drives, 
- [ ] General cleanup
- [ ] Localisation support? UI strings are very different in TS1, and the ones custom for simitone are hardcoded.
- [ ] Some rearrangement for "desktop" class screens. Touchscreen targeted UI will still be present.
- [ ] Picture in Picture. 

## Neighborhood Management
- [ ] Lot Bulldoze (and move out)
- [ ] Lot Rezone
- [ ] Neighbourhood Bootstrap (eg. works without opening ts1 once)
- [ ] Neighbourhood Management Page
  - [ ] Copy Neighbourhoods from TS1
  - [ ] Create new named neighbourhood (from scratch)
  - [ ] Delete/Rename neighbourhoods.

## Engine Features
- [ ] Promote/Demote Fame Logic
- [ ] Accurate Free Will. Need a better handle on how this works in TS1.
  - [ ] Self modification of Motive Advertisements via the test function.
- [ ] Saving the game when on Vacation.
- [ ] Autofollow sim plumbbob. 
- [ ] Some way for the Simitone plumbob to show motive state. (likely in the diamond)
- [ ] Purchasable outfits and saving back to the character IFFs.
- [ ] All TS1 global calls, including the stupid ones like Pet to Adult.

## QA
Note: This also includes things that _should_ work with the current Simantics/engine feature set, but probably doesn't and may need further investigation.
- [ ] Verify Core Gameplay
  - [ ] Career Progression
  - [ ] Child. School, military school, studying.
  - [ ] Relationship cycle. Walkby -> phone invites -> romance -> propose -> move in -> baby -> child
  - [ ] Death (adult, child, pet)
- [ ] Verify Downtown Gameplay
  - [ ] Restaurant lifecycle
  - [ ] Purchasable Clothing
  - [ ] Auto-Follow, aka "invite sim to downtown" or "take sim home"
  - [ ] Find out what's wrong with those god damn cars
- [ ] Verify Vacation Gameplay
  - [ ] Saving game on vacation. Other families cannot visit a vacation lot with another sim present. Going to a vacation lot someone was in before cleans it up (removes all npcs, resets all objects)
  - [ ] Renting Stuff. Igloo, snowman kit, picnic basket etc.
  - [ ] "Vacation Score", a cross lot score that gets you a prize
  - [ ] Souvenirs
  - [ ] Hotel booking. At time of issue creation the game always thinks you are booked in... it probably dynamically switches the player sim between visitor and roommate status.
- [ ] Verify Unleashed Gameplay
  - [ ] Purchase Pets.
  - [ ] Pet AI. (shouldn't randomly starve to death, at least)
  - [ ] Pet shows and training.
  - [ ] Lifecycle for all object pets.
  - [ ] Renting stuff, like the boat. I think this is similar to vacation's.
- [ ] Verify Studiotown Gameplay
  - [ ] Fame career track
  - [ ] Fame special objects, like skydiving, aquarium, satellite dish
  - [ ] Obsessed fan, studio tours visit (at home)
- [ ] Verify Makin Magic Gameplay
  - [ ] Living in Magic Town via the houses purchasable with Magicoins
  - [ ] Quests
  - [ ] Purchasable Items
  - [ ] Spells
  - [ ] Craftable items
  - [ ] Pyrite (right now it doesn't seem to do anything autonomously?)
- [ ] Object CC. Iff and Far, with custom anims and maybe sounds
- [ ] Skin CC

# Would like to have:
- [ ] Simitone Website
## FreeSO Engine Improvements (can be delayed)
- [ ] 2D 2.0: Less stuttering
  - [ ] Object sprite partial vertex buffer update, rather than full regeneration every change.
  - [ ] 2D Walls rewrite, likely with same theory as above.
  - [ ] Cleaner no-depth implementation from the start? We can make nicer switchable shaders with compiler definitions and imports.
- [ ] 3D 2.0: Additional graphics features for remeshes.
  - [ ] Import normals from .OBJ. 
  - [ ] Cube map and specular tools.
  - [ ] Investigate options for normal maps and specular maps (with texture compression)
  - [ ] New volcanic tools
    - [ ] Layer management. Reorder layers within a 3d drawgroup, cull mode, lighting parameters.
    - [ ] Mesh Replacements -> .far archive with texture compression. (DXT, ETC, PVRTC) Good for hitting low spec systems.

# Possible future features:
- [ ] 3D Neighbourhoods (via replacement model. game can already export a base model for ngbh that creators can line up properties with)
- [ ] Family Import/Export (maybe compatible with original)
- [ ] House Import/Export (NOT compatible with original)
- [ ] Simitone Neighbourhood Import/Export (likely just a copy, though might want to verify expansions and CC)
- [ ] Lighting redux
  - [ ] Shadow mapped outdoor light with cascades. Will allow for object self shadow outdoors, sharp shadows near camera, direct outdoors lighting decoupled from light sources.
  - [ ] Smarter localized repainting for lights. When an object moves, only redraw near it. When a light appears/disappears, only redraw within its range.
  - [ ] Make lights cast from midpoint of their bounding box, rather than always the ceiling. (we have estimated bounds in 2d now as well)
  - [ ] Remove room limit for lights in ultra lighting mode. Lights will shine directly through windows, casting shadows with the window shape. Old light bleed will be made very slight so it appears like "bounce lighting".
- [ ] Hints?

**Comment by CitrusAlex** [comment]
Would it be possible to add Linux version to "
Would like to have"?

**Comment by riperiperi** [comment]
Technically it already works:
http://forum.freeso.org/threads/freeso-on-linux-natively.6249/

It's not a focus for me, but I suppose porting the FreeSO style dynamic linking of DirectX vs OpenGL should be done as well.

**Comment by francot514** [comment]
Skill progress reporting, and value rate adjust should be also included into list.

**Comment by MrPowerGamerBR** [comment]
@Patrik356b kinda late to the conversation: As far as I know, EA itself told riperiperi to discontinue the Simitone's Android port. (fuck them)

http://freeso.org/ts1-ios-port-cancelled/

Even then, I still hope Simitone works as a replacement for the Sims 1 client, The Sims 1 works very wonky on Windows 10 and never worked under Linux (via WINE), so having an open source client for it would be awesome

**Comment by tannisroot** [comment]
@MrPoweGamerBR it actually works perfectly with dgvoodoo2. You can try out Lutris installer to see yourself.

**Comment by Patrik356b** [comment]
> @Patrik356b kinda late to the conversation: As far as I know, EA itself told riperiperi to discontinue the Simitone's Android port. (fuck them)
> 
> http://freeso.org/ts1-ios-port-cancelled/

You do realize that I recently found these projects, right? 

EA knows that they can't deliver a good experience, hence they try to stop anything. Assholes...

**Comment by MrPowerGamerBR** [comment]
@tannisroot I actually found out about Lutris after I commented that! I didn't want to try Lutris because I thought it wouldn't work (because The Sims is set to garbage on winedb) but actually it works pretty amazingly with Lutris! (As long as your graphic card supports Vulkan, since it uses DXVK, while the default Wine doesn't and has a bunch of graphic glitches)

**Comment by johnchau95** [comment]
I would like all of the features on riperperi’s list for Simitone plus:
- An expanded Downtown, Vacation Island, Studio Town and Magic Town, expanded from their original sizes to the size of Old Town, with additional surrounding lots. Each lot can now be rezoned so Sims can now live in Downtown and Studio Town and rent or purchase vacation homes at Vacation Island.
- Cheering animations for pets (may finally allow Vacation Director Kana to give a reward to families with pets).
- Shooing animations for children shooing away El Bandito, previously missing from Unleashed.
- When playing a lot, surrounding areas of the neighbourhood including lots, trees, rocks, lakes, bridges and beaches should also be rendered.
- More realistic ponds and pools and water terrain tools for lots. The original water terrain tools simply splattered a flat layer of water over grass and rendered pools that look like flat blue floors with no water animation whatsoever. Both look even worse in 3D mode. For the new version, the water terrain tools should actually dig down the terrain and render a more realistic water pond in that position, with rippling animation for the water and a reflection for adjacent Sims, objects, walls, ceilings, roofs and the sky with the sun, moon, stars and clouds, and render pools that have actual pool walls and tiles and a translucent water animation with real ripples and splashes and a similar reflection. Clearly the original ponds in the lots and the pool ladder will have to be completely redone for this release.
- The water in the hot tubs should be animated even when they are not being used, with constant rippling and a reflection for the top part of the tub as well as adjacent Sims, objects, walls, ceilings, roofs and the sky with the sun, moon, stars and clouds. Same goes with the nectar press.
- A slightly expanded Old Town neighbourhood view adapted for widescreen and Retina displays, expanding from the original 4:3 area to 16:10 by adding in more information to the left, right, top and bottom sides of the image (also restoring the detail lost in converting from the original neighbourhood view to Old Town in Unleashed). This neighbourhood mode should also be rendered (with four rotations as in lot mode) to look more realistic, with smoother and more lifelike animation (especially in the waterfalls), actual traffic in the roads (not just on the highway of Studio Town), variably stopping at junctions and turning, more realistic lighting and shadows of the lots, trees, rocks and traffic, and sea water that is less deep blue, with reflections of the sky, buildings and bridges.
- A new optional open world feature similar to The Sims 3 and 4, where Sims can now freely explore the neighbourhood they’re in and walk or jog to any location as well as visit other Sims’ houses. Old Town, Downtown, Vacation Island, Studio Town and Magic Town are effectively combined into one “Maxis Island” (similar to the layout here: [](https://www.youtube.com/watch?v=_tlgB6oSYas))
  - Locations previously inaccessible in neighbourhoods are now accessible:
  - The lakes are now fishing spots; Sims can now fish at the edge of the cliff or on the bridge above the lake.
  - Sims can now actually swim in the sea of the beaches of Downtown and Vacation Island as well as the originally inaccessible Old Town Beach. The water on the beaches should look more realistic, with waves crashing down on the sand and receding every few minutes.
  - The tunnel in Old Town is now a rabbit hole. It can take Sims to work or school or to Studio Town or Magic Town.
  - In Studio Town, Sims can now watch movies and attend red carpet events at the Cinema, go sightseeing and watch or hold concerts at the Concert Hall.
  - At Vacation Island, the ferry port is now an accessible location. Sims can now use the ferry to take a cruise trip around Maxis Island for a determined period of 3, 5, 7, 10 or 14 days or take a 6-hour trip back to the ferry port at Old Town Beach or near SimCity Beach in Downtown, where they can also take a 6-hour ferry trip to Vacation Island or go on a 3-hour sightseeing boat trip around Old Town and Downtown.
  - At the Downtown police station, Sims can now view incarcerated criminals in their jail cells (sometimes other Sims from Maxis Island in the Criminal or Psychic career track or with a Magic wand or Obsessed Fans) or ask the monitoring police officers for advice on how to be a good citizen or starting out in the Law Enforcement career track. If these police officers come from existing families in Maxis Island, Sims can even build a relationship with them and ask them what they think about their career. Families with at least one kid in military school can ask the officer to help them bail the kids out so they can come home, and Sims who know certain incarcerated criminals can negotiate a paid amount of Simoleons to bail them out. The chance will be higher if these Sims are friends with the officer, officers who are enemies with the Sims will most likely refuse the request. There is also a 1/3 chance that any kid who is sent home from military school will come back as an adult in the Military career track with a hostile relationship with their parents. They may even refuse to come home to their families and if so will then be found in the Family Bin so they can move into a new home.
- New Sunrise/Day/Sunset/Night Mode for neighbourhood view. The lights of the lots can be switched on and off and the headlights of the vehicles can also light up.
- Sims now build Cooking and Mechanical skills when cooking meals and repairing objects.
- The Fame career is now entirely skill-based and no longer depends on specific orders. Combinations are still present, yet they no longer influence the overall result, rather how successful the Sims will be solely depends on how high their Charisma, Logic and Creativity skills are. Sims who have maxed all three skills will always be successful.
- Sims can now call in sick in order to get a paid day off from work. If they are genuinely sick, they will always be successful; otherwise how successful they will be depends on their job performance and Sims may get caught faking it. They can also quit their job over the phone.
- Sims can now use the newspaper or the computer to find a new place to live and move out from their current household, taking any other Sims they like with them.
- Get A Life Mode from The Sims for console to be ported as SimValley, an uptown district of SimCity located a bridge across from Maxis Island. A Sim can be created to play the Get a Life Mode from beginning to end. Every reward can be unlocked as the game progresses with no time limits on any rewards, and once the final level is complete, the Sim can still be played and the whole of SimValley as well as all the families can be played.
- Bust Out Mode from The Sims Bustin’ Out to be ported as WillVille, another uptown district of SimCity located a bridge across from Maxis Island. A Sim can be created to play the Bust Out Mode from beginning to end. Every reward can be unlocked as the game progresses with no time limits on any rewards, and once the final level is complete, the Sim can still be played and the whole of WillVille as well as all the families can be played. Plus the ability for all Sims to purchase, own and drive cars is unlocked.
- Previously missing animations for vehicles arriving and car doors are restored, allowing you to see the Sims getting on the ride to work, school or any other location, as well as the vehicles arriving on the lot and driving to the location (with steering animations). In addition, transportation services are no longer limited by location and players can select how many Sims to go on the transport.
- Objects, walls and floors from The Sims console series (The Sims and The Sims Bustin’ Out) to be adapted and ported directly to this release.
- Completely new 2D and 3D sprites for every object, wall and floor from and downloadable for The Sims Complete Collection, The Sims console series and The Sims Online, upscaled to 16x the resolution (4length x 4width) for slightly more realistic and detailed objects.
- Every Sim skin from and downloadable for The Sims Complete Collection and The Sims Online to be upscaled to 16x the resolution (4length x 4width) for slightly more detailed-looking Sims, with extra separate switchable and animated layers for eyes (for blinking and sleeping) (eye color changes only in Create-a-Sim), teeth (slight mouth movement) (changes only in Create-a-Sim), hats (removed when nude or wearing athleticwear, sleepwear or swimwear) glasses (removed when nude or wearing sleepwear or swimwear), hair (colour and style), and makeup, and bump maps for clothing and five-fingered hands. The Sims Creator tool should also be updated to reflect this.
- The Neighbourhood 2 from Livin’ Large/Livin It Up will effectively be merged into Neighbourhood 1, meaning that the Mashuga family will now live in the original Maxis Island at 9 Sim Lane. The house that would’ve replaced the Newbie House at 7 Sim Lane in Neighbourhood 2 will now be located at 4 Sim Lane.
- Starting a new neighbourhood should start a completely new empty neighbourhood, Downtown, Vacation Island, Studio Town and Magic Town (or at least a completely empty Maxis Island which doesn’t add in the lots and families from Old Town and the other districts).
- Roofs no longer spawn by default once a room has been fully built. They can now be built in custom locations, sizes and styles.
- New roof colours and styles from The Sims 2 and above can now be built into the houses, some with walls of their own.
- Lots and houses no longer have a two floor limit, allowing for taller buildings.
- The Simitone UI to be disabled by default, instead using the original UI from The Sims by default (optimised for widescreen and Retina displays). The Simitone UI can be re-enabled later or switched to a less dark blue and more light purple version of the UI from The Sims, similar to The Sims Online.
- Every song from the soundtrack of The Sims and The Sims Bustin’ Out to be replaced with higher quality transfers redelivered from higher quality sources and delivered as 320kbps MP3, 528kbps or 576kbps variable bit AAC M4A or lossless WAV, AIFF or FLAC files.
- If possible, a slight remix of the ambience, sound effects and music soundtracks and a remap of the sound engine to allow for 5.1 surround sound output via SPDIF or compatible sound card.
- The Options Menu can now be accessed in Neighbourhood View and Free Will can be set globally for all houses. In addition, players can now switch between 2D and 3D mode in-game rather than add the “-3d” extension to the shortcut to start the game in 3D mode.
- The original intro video for the game, removed from Superstar and later for unexplained reasons, should now be restored when you load the game. Options to select which EA and Maxis logos, splash screens and loading taglines you want to appear should also be available.
- Save files, rather than being saved in the program files, should be saved in the My Documents folder, allowing for separate save files for each user, and a refresh by deleting the saved files allowing the game to generate fresh ones upon the next load.
- A fix for lighting, especially moving lights with multicoloured sections. Normally, in game, lights with moving bulbs of different colours leave a spotlight on their own surface, but not the floor or the adjacent walls, objects or Sims. This should be fixed so that the lights leave spotlights on the floor and adjacent walls, objects and Sims. Ceiling lights with rotating fans should be able to have a rotating shadow dissipating across the room as well.
- A 32-bit version with the above changes which maintains direct compatibility with Windows XP Service Pack 3 and DirectX 9.0c (and Windows 98SE/NT4/2000/ME if original game or Deluxe or Double Deluxe (and any additional expansions up to Makin’ Magic) are installed) and does not require the FreeSO client, instead using a separate standalone installer. (Maybe it would be best to start the 32-bit version as a separate build on a Windows XP PC or virtual machine with DirectX 9.0c to implement the FreeSO lighting engine plus the additional features and fixes listed but not include a 3D mode for the time being, just for the sake of maintaining compatibility.)

**Comment by dotequals** [comment]
@johnchau95 that is quite the list! I can speak to a few points, but most importantly, I should emphasize that Simitone uses FreeSO’s engine so the benefits there will be brought over and many already have! If you want to find out more information or chat more about these topics, I’d recommend joining our [Discord](https://discord.gg/xveESFj).

- missing animations: if they’re in the game files, patches (piffs) could be created that include those trees. If suitable options aren’t in the game files, custom animations would need to be made. This is a relatively new feature for FreeSO, but you can find out about [Importing/Exporting animations on the FreeSO wiki](https://github.com/riperiperi/FreeSO/wiki/Avatar-Tool:-Exporting-and-Importing-Animations).
- pool/water: pool tiles were just given an update in FreeSO (version 78h) so this will be brought over
- skilling from cooking, etc: this would work better as custom content or piffs as not every one will want this experience and it will be able to be done with Volcanic (you can use Volcanic with Simitone by starting Simitone with the flag `-ide`, but the publicly available release is a bit annoying to read tree text in.
- more than 2 stories: both Simitone and FreeSO support lots having up to 5 floors.
- Simitone UI: the non-touch UI is already underway and it’ll scale much better than the existing graphics
- options/2D/3D mode: this is a rather new change to FreeSO, but the `-3d` flag now means enable 3D features and hybrid 2D and with this option you can toggle 2D/3D via F12 or a setting in the options menu. Without the flag, the game runs in exclusive 2D which is great for older computers.
- game saves: the goals of FreeSO and Simitone are to never modify the original games’ files. You’ll see this already in action after launching Simitone for the first time you’ll have a Simitone folder in Documents containing the save files it uses
- lighting: Most of this is covered with FSO lighting and the ability to color lights will also likely be introduced.
- Windows XP: This is outside the scope of the project. FreeSO and Simitone use .NET 4.6 which requires Windows Vista SP2 and Volcanic uses 4.6.1 which requires Windows 7 SP1. If the build does not currently support 32-bit systems, it will in the future. If you’re running Windows XP, you’re better off using the original Sims.exe.
- images/videos/audio/objects/EA’s intellectual property: **we can not and will not distribute any of EA’s IP.** If you own something and want to put it in your game that’s fine, but they cannot be committed to the repository.

**Comment by riperiperi** [comment]
This issue is a roadmap for the first releasable version, not a wishlist. If those features were to be considered, they would be after the list here has been completed, and potentially part of another future series of issues and milestones.

As for neighborhood management, the idea is to have generation of neighorhoods controllable via a menu rather than just 1-9, which will allow you to generate the sub areas with blank lots instead, and more than 9 neighborhoods (each named rather than numbered).

**Comment by johnchau95** [comment]
Quick question: if anyone wanted to add the new FreeSO lighting and shadow engine on its own to any version of The Sims (original, Deluxe, any expansions or Complete) on Windows 98/2000/XP/7/8/10, as a seperate add-on, mod or patch, without having to go through the trouble of downloading Simitone, would the creation of this standalone download be possible?

**Comment by tannisroot** [comment]
@johnchau95 that's not something that can be easily ported over from FreeSO and added as a mod, the engine implementations are completely different.

**Comment by gMagnus87** [comment]
Any news about the next Update? 🤔
It's been a year since the last update, I still can't use it.

**Comment by Sagatt** [comment]
> 
> 
> Any news about the next Update? 🤔
> It's been a year since the last update, I still can't use it.

Project has been shutdown by E.A if i'm not mistaken

**Comment by MrPowerGamerBR** [comment]
@Sagatt no, the iOS port was cease and desist'd by EA. The Windows port is still alive, however it is kinda obvious that the cease and desist made riperiperi unmotived to invest more time on this project, and I don't blame him.

Keep in mind that *technically* riperiperi didn't do anything wrong and EA just made the cease and desist because "lol we don't wanna to lose our The Sims Mobile sales so bye", but he compiled to avoid repercurssions on the FreeSO project.

**Comment by IntelMiner** [comment]
It looks like they're still active on FreeSO. Would it be possible to tag them directly to get an update?

This project is fascinating and I'd love to see it achieve the same heights as OpenRA, OpenMorrowind and others

@riperiperi

**Comment by SkyNinja** [comment]
Oof, not just abandoned, but actively ignoring this project? Damn, that sucks.

**Comment by riperiperi** [comment]
I suppose it does suck. Feel free to spend your own time continuing the project, if you would like it to move faster.

**Comment by IntelMiner** [comment]
>if you would like it to move faster.

Does this suggest that the project may get picked up again at some point in future? 

Obviously none of us can suggest how you use your time as a developer. Most of us just want to know if the project is on hiatus or has indeed been shut down by EA

**Comment by SkyNinja** [comment]
You have every right to work on what you want, but that doesn't change the fact that having the only way to play a classic game on modern systems without putting your whole trust in [this](https://www.virustotal.com/gui/file/6e598c22512a24659cbd0e533306abf3129733d67b1d120916b1b6b1581e29f2) be in a state of total ambivalence _does suck._

**Comment by ShaunRoselt** [comment]
@riperiperi Is this list up to date?

Because I see the last edit was in 2019.

Would you mind updating it please if it's not up to date? 🙏🏻

**Comment by tannisroot** [comment]
> [@riperiperi](https://github.com/riperiperi) Is this list up to date?
> 
> Because I see the last edit was in 2019.
> 
> Would you mind updating it please if it's not up to date? 🙏🏻

AFAIK the project is dead

**Comment by salvadorc17** [comment]
Yes sure list need to be updated, but you can check the progress here:
https://github.com/alexjyong/Simitone
